### PR TITLE
[tree] Fix invalid macro names when absolute filepath is passed

### PR DESCRIPTION
**Pull request by ferdymercury**
Fixes https://github.com/root-project/root/issues/16562

### CODE REVIEW

File: tree/treeviewer/src/TTreeViewer.cxx [hunk 1]
```diff
@@ -2741,10 +2743,8 @@ void TTreeViewer::SaveSource(const char* filename, Option_t *)
    }
    //   Write macro header and date/time stamp
    TDatime t;
-   TString sname(fname);
-   sname = sname.ReplaceAll(".C", "");
    out <<"void open_session(void *p = 0);"<<std::endl<<std::endl;
-   out <<"void "<<sname.Data()<<"() {"<<std::endl;
+   out <<"void "<<std::filesystem::path(fname.Data()).stem()<<"() {"<<std::endl;
```
Comment (pcanal) [inline tree/treeviewer/src/TTreeViewer.cxx:2747]:
```suggestion
   out <<"void "<<std::filesystem::path(fname).stem()<<"() {"<<std::endl;
```

**Comment by guitargeek** [comment]
@ferdymercury, the test failure on alma8 is real

**Comment by ferdymercury** [comment]
> @ferdymercury, the test failure on alma8 is real

Good catch. Solved now.

**Review by guitargeek — APPROVED** [review]
Thanks a lot, looks good now!

**Pull request merged**